content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -173,22 +175,34 @@ class CreateAndIssueMarsVoucherInitiator @JsonConstructor constructor(private va
     lateinit var jsonMarshallingService: JsonMarshallingService
 ```
 
-#### Add the flow implementation
+#### Write your business logic
+
+##### Add the flow implementation
 
 Next you must add the flow implementation to the initiating flow by encapsulating it within the `call` method.
 
 1. Add the `@Suspendable` annotation.
 2. Add the `call` method with the return type `SignedTransactionDigest`.
 3. Parse your parameters using the `mapOfParams` value. Since the parameters the flow will use come in JSON format, the JSON object must be parsed to extract the parameters so the flow can run.
-4. Add exceptions for when required fields (`voucherDesc`, `holder`, `recipientParty`) are not found.
-5. Insert a method for finding the notary.
+4. Add appropriate error handling - exceptions must be thrown when required fields (`voucherDesc`, `holder`, `recipientParty`) are not found.
+5. Insert the following method for finding the notary: `notaryLookup.getNotary(CordaX500Name.parse("O=notary, L=London, C=GB"))!!`.
```

Comment:
might want a breakout box that mentions what notary to find etc

Reply:
@peterli-r3 could you pls specify this one?

Reply:
The notary in Corda 5 DP1 was pre-defined. You can find the full status of the notary by start the network and print out the status of the network.

Reply:
That's too literal @peterli-r3 

We should discuss a little what this means, i.e. we will be "spending" our state with a particular notary tieing that state to it (notary change excepted). thus, we can't just "pick the first one" as the list order may change, especially in a multi-network / app environment where different notaries are whilsatestied for different apps. 

We are tryign to teach people how to write an app, not duplicate the DP example

Reply:
added a note. @peterli-r3 and @fenryka please review